manifests/prepare/stringify_facts.pp

```diff
@@ -0,0 +1,21 @@
+# == Class puppet_agent::prepare::stringify_facts
+#
+# A preperation class to configure the stringify_facts setting to false
+#
+class puppet_agent::prepare::stringify_facts {
+
+  if (versioncmp($::clientversion, '4.0.0') < 0) {
+
+    ini_setting { 'puppet stringify_facts':
+      ensure  => present,
+      path    => $::puppet_config,
+      section => 'main',
+      setting => 'stringify_facts',
+      value   => false,
+    }
+
+  } else {
+    fail('The puppet_agent::prepare::stringify_facts class should only be run on Puppet < 4')
```

Comment:
This approach is slightly awkward, because it means you have to apply this class, then remove it and apply puppet_agent to upgrade. Letting it be a warning might be slightly friendlier.

Reply:
Yeah, I thought the stringify_facts setting would throw an error with Puppet 4, but it's just ignored, so I'll switch this to a warning 👍